oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/oidc/authentication/OidcClientRegistrationAuthenticationProvider.java

```diff
@@ -222,7 +223,8 @@ private OidcClientRegistrationAuthenticationToken registerClient(OidcClientRegis
 	private OAuth2Authorization registerAccessToken(RegisteredClient registeredClient, String issuer) {
 		JoseHeader headers = JwtUtils.headers().build();
 
-		Set<String> authorizedScopes = Collections.singleton(DEFAULT_CLIENT_CONFIGURATION_AUTHORIZED_SCOPE);
+		Set<String> authorizedScopes = Collections.unmodifiableSet(
```

Comment:
This could be changed to:

```java
		Set<String> authorizedScopes = Collections.unmodifiableSet(
				Collections.singleton(DEFAULT_CLIENT_CONFIGURATION_AUTHORIZED_SCOPE));
```

to reduce the extra `HashSet` creation but it may be confusing passing an immutable set to `Collections.unmodifiableSet()`. I think it would less confusing with:

```java
Set<String> authorizedScopes = new HashSet<>();
authorizedScopes.add(DEFAULT_CLIENT_CONFIGURATION_AUTHORIZED_SCOPE);
authorizedScopes = Collections.unmodifiableSet(authorizedScopes2);
```

What do you think?

Can you also please add a test. Thanks.

Reply:
Sure @jgrandja , I'll do the change. 
I can change `authenticateWhenClientRegistrationRequestAndValidAccessTokenThenReturnClientRegistration` test to check if the authorized scope is unmodifiable. 

There is already an assertion checking that only one scope is available. 
```
assertThat(authorizationResult.getAccessToken().getToken().getScopes())
				.containsExactly("client.read");
```

I will assert that UnsupportedOperationException is thrown when a new element is added to the Set (it can be changed to `isUnmodifiable` from assertj when the version will be upgraded to `3.21.0`)